packages-rs/drainpipe/src/db.rs

```diff
@@ -27,13 +29,20 @@ pub fn update_seq(conn: &mut SqliteConnection, new_seq: i64) -> anyhow::Result<(
 
 pub fn record_dead_letter(
     conn: &mut SqliteConnection,
+    kind: ProcessErrorKind,
     new_msg: &str,
     new_seq: i64,
+    message: &str,
```

Comment:
should this type instead be a `Vec<u8>`, and then do the `String::from_utf8` here? I'm thinking about matching the interface with the `kind` argument, which has an internal type that gets converted to a db type.  I think either they should both be db types (`i32` and `&str`), or both be internal types (`ProcessErrorKind` and `Vec<u8>`) and get converted here. 

Only other thing to consider is the possibility of `String::from_utf8` returning an Err, but I'm not sure what would cause that.  There's also a `from_utf8_unchecked` that might make sense to use here? not sure!

Reply:
I think in general it's ok for the contents to be best effort. The practical use case for the dlq involves re-consuming those sequences from the firehose. This message is more for quick historical debugging.

Reply:
And yeah I think performing the conversion inside of this function, ideally with a `Serialize` trait from diesel if that exists (see other comment), is the way to go here.